pkg/network/ebpf/c/protocols/classification/routing-helpers.h

```diff
@@ -0,0 +1,81 @@
+#ifndef __PROTOCOL_ROUTING_HELPERS_H
+#define __PROTOCOL_ROUTING_HELPERS_H
+
+#include "ktypes.h"
+#include "protocols/classification/defs.h"
+
+static __always_inline bool is_last_program_from_layer(classification_prog_t current_program) {
+    classification_prog_t next_program = current_program+1;
+    if (next_program == __PROG_APPLICATION ||
+        next_program == __PROG_API ||
+        next_program == __PROG_ENCRYPTION ||
+        next_program == CLASSIFICATION_PROG_MAX) {
+        return true;
+    }
+
+    return false;
+}
+
+// The following macros return true in case where no BPF programs are registered for
+// a given layer or false otherwise.
+#define is_application_layer_empty() is_last_program_from_layer(__PROG_APPLICATION)
+#define is_api_layer_empty() is_last_program_from_layer(__PROG_API)
+#define is_encryption_layer_empty() is_last_program_from_layer(__PROG_ENCRYPTION)
```

Comment:
do you really need these helpers? why not use the `is_last_program_from_layer`

Reply:
Done